### PR TITLE
Test Billing Sals

### PR DESCRIPTION
**Pull request by MElborolossy**
### Description

Add Tests for billing

### Changes

- [x] submit_payment
- [x]  wait_payment
- [x]  refund_failed_payments
- [x]  issue_refund
- [x] issue_refund_with_specific_amount
- [x]  check_refund
- [x] check_refund_raise_error
- [x]  process_refunds
- [x]  process_payments
- [x]  refund_extra
- [x] refund_extra_set_false

### Related Issues

#2140

### Checklist

- [x] [Pre-commit hook is installed](https://github.com/threefoldtech/js-ng#pre-commit) to do formatting checks before committing code...etc
- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

**Comment by MElborolossy** [comment]
All test passed
image

File: tests/sals/billing/test_billing.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+        self.assertTrue(j.sals.billing.wait_payment(payment_id), "This payment must be success")
+        self.info("Issue refund for the success payment to source wallet")
+        j.sals.billing.issue_refund(payment_id)
+        j.sals.billing.process_refunds()
+        self.info("Check if the source wallet refunded correctly")
+        # Get balance after refund
+        balance_after_refund = self.test_source_wallet.get_balance_by_asset()
+        # Check if the difference between two balances is 0.2, as 0.1 will used as transaction fees for every single transcation
+        self.assertAlmostEqual(balance_before_transfer - balance_after_refund, 0.2)
+        pass
```
Comment (waleedhammam) [inline tests/sals/billing/test_billing.py:168]:
```suggestion
        
```

File: tests/sals/billing/test_billing.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+        self.assertTrue(j.sals.billing.wait_payment(payment_id), "This payment must be success")
+        self.info("Issue refund for the success payment to source wallet")
+        j.sals.billing.issue_refund(payment_id, amount=refund_amount)
+        j.sals.billing.process_refunds()
+        self.info("Check if the source wallet refunded correctly")
+        # Get balance after refund
+        balance_after_refund = self.test_source_wallet.get_balance_by_asset()
+        # Check if the difference between two balances is 0.2, as 0.1 will used as transaction fees for every single transcation
+        self.assertAlmostEqual(balance_before_transfer - balance_after_refund, refund_amount + 0.2)
+        pass
```
Comment (waleedhammam) [inline tests/sals/billing/test_billing.py:207]:
```suggestion
        
```

**Review by waleedhammam — APPROVED** [review]

File: tests/sals/billing/test_billing.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+    def test_09_process_payments(self):
+        """Test Case for process payments
+
+        **Test Scenario**
+        - Submit a payment with amount of 1 TFT to destination wallet
+        - Check that payment is submitted and success state False
+        - Transfer 1 TFT from source wallet
+        - Run process_payments
+        - Check that payment success state True
```
Comment (AhmedHanafy725) [inline tests/sals/billing/test_billing.py:284]:
```suggestion
        """Test Case for processing payments.

        **Test Scenario**
        - Submit a payment with an amount of 1 TFT to the destination wallet.
        - Check that payment is submitted and success state False.
        - Transfer 1 TFT from the source wallet.
        - Run process_payments.
        - Check that payment success state True.
```

File: tests/sals/billing/test_billing.py [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+    def test_10_refund_extra(self):
+        """Test case for refund extra payments
+
+        **Test Scenario**
+        - Submit a payment with amount of 1 TFT to destination wallet
+        - Get source wallet balance
+        - Transfer 2 TFT from source wallet
+        - Run refund_extra
+        - Get source wallet balance
+        - Check if refund_extra return correct amount
```
Comment (AhmedHanafy725) [inline tests/sals/billing/test_billing.py:315]:
```suggestion
        """Test case for refunding extra payments.

        **Test Scenario**
        - Submit a payment with an amount of 1 TFT to the destination wallet.
        - Get the source wallet balance.
        - Transfer 2 TFT from the source wallet.
        - Run refund_extra.
        - Get the source wallet balance.
        - Check if refund_extra returns the correct amount.
```

File: tests/sals/billing/test_billing.py [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    def test_11_refund_extra_set_false(self):
+        """Test case for refund extra payments and set refund_extra False
+
+        **Test Scenario**
+        - Submit a payment with amount of 1 TFT to destination wallet with refund_extra flag equals False
+        - Transfer 2 TFT from source wallet
+        - Get source wallet balance
+        - Run refund_extra
+        - Get source wallet balance
+        - Check if balance not changed
```
Comment (AhmedHanafy725) [inline tests/sals/billing/test_billing.py:348]:
```suggestion
        """Test case for refunding extra payments and set refund_extra False.

        **Test Scenario**
        - Submit a payment with an amount of 1 TFT to destination wallet with refund_extra flag equals False.
        - Transfer 2 TFT from the source wallet.
        - Get the source wallet balance.
        - Run refund_extra.
        - Get the source wallet balance.
        - Check if the balance not changed.
```

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Comment by MElborolossy** [comment]
After last commit all tests passed
image

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**